units/pod_initializing.go

```diff
@@ -87,11 +86,18 @@ func (j *podInitJob) Run(ctx context.Context) {
 
 	switch j.pod.Status {
 	case pod.StatusInitializing:
-		if _, err := j.ecsClient.RunTask(ctx, &ecs.RunTaskInput{
-			Cluster:        utility.ToStringPtr(j.pod.Resources.Cluster),
-			ReferenceId:    utility.ToStringPtr(j.PodID),
-			TaskDefinition: utility.ToStringPtr(j.TaskID),
-		}); err != nil {
+		execOpts := cocoa.NewECSPodExecutionOptions().
+			SetCluster(j.pod.Resources.Cluster).
+			SetExecutionRole(j.env.Settings().Providers.AWS.Pod.ECS.ExecutionRole)
```

Comment:
I think you have to rebase main because this is outdated - this was moved to the creation options.